PolyEngine/Engine/Src/HeavyTaskSystem.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "HeavyTaskImplementation.hpp"
+#include "World.hpp"
+
+namespace Poly
+{
+	namespace HeavyTaskSystem
+	{
+		void HeavyTaskPhase(World* w);
+
+		inline UniqueID SpawnEntityImmediate(World* w)
```

Comment:
We should move some of these implementation to cpp file. (only non-template ones)

Also for immediate task there is no need to create HeavyTask objects. Just execute the task on world.